contracts/core/EventEmitter.sol

```diff
@@ -0,0 +1,44 @@
+// SPDX-License-Identifier: AGPL-3.0
+pragma solidity ^0.8.15;
+
+import { Call } from "./types/Common.sol";
+
+/**
+ * @title Event Emitter
+ * @notice Is responsible for emitting events
+ * @dev Allows us to emit events in the context of a single contract
+ * rather than events being emitted in the context of every users
+ * proxy contract.
+ */
+contract EventEmitter {
+  /**
+   * @dev Emitted once an Action has completed execution
+   * @param name The Action name
+   * @param proxyAddress The proxy address of the user executing the t/x
+   * @param returned The bytes value returned by the Action
+   **/
+  event Action(string name, address proxyAddress, bytes returned);
```

Comment:
`indexed` the name at least. Not sure about `proxyAddress`, I have no idea what's the purpose behind that.

Reply:
ProxyAddress is a requirement from Damian so it's easy to map back events to a given user.